src/openrct2-ui/input/Shortcuts.cpp

```diff
@@ -192,7 +192,8 @@ static void ShortcutAdjustLand()
 
     if (!(gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR) || gEditorStep == EditorStep::LandscapeEditor)
     {
-        if (!(gScreenFlags & (SCREEN_FLAGS_TRACK_DESIGNER | SCREEN_FLAGS_TRACK_MANAGER)))
+        if (!(gScreenFlags & (SCREEN_FLAGS_TRACK_DESIGNER | SCREEN_FLAGS_TRACK_MANAGER))
+            && (!(gParkFlags & PARK_FLAGS_FORBID_LANDSCAPE_CHANGES) || gCheatsSandboxMode))
```

Comment:
Instead of checking `gCheatsSandboxMode`, you'll also need to check if the player is in the scenario editor. This is another screen flag, I believe.